pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -679,35 +679,10 @@ func (og *operationGenerator) GenerateMountVolumeFunc(
 			resizeOptions.DeviceStagePath = deviceStagePath
 		}
 
-		// No mapping is needed for hostUID/hostGID if userns is not used.
-		// Therefore, just assign the container users to host UID/GID.
-		hostUID := util.FsUserFrom(volumeToMount.Pod)
-		hostGID := fsGroup
-		if utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesStatelessPodsSupport) {
-			// Without userns hostUID/GID was the user inside the container too.
-			containerUID, containerGID := hostUID, hostGID
-
-			kvh, ok := og.GetVolumePluginMgr().Host.(volume.KubeletVolumeHost)
-			if !ok {
-				msg := fmt.Errorf("volume host does not implement KubeletVolumeHost interface")
-				eventErr, detailedErr := volumeToMount.GenerateError("MountVolume type assertion error", msg)
-				return volumetypes.NewOperationContext(eventErr, detailedErr, migrated)
-			}
-
-			// This pod _might_ use userns. GetHostIDsForPod() will give us the right
-			// UID/GID to use for this pod (no matter if the pod uses userns or not).
-			hostUID, hostGID, err = kvh.GetHostIDsForPod(volumeToMount.Pod, containerUID, containerGID)
-			if err != nil {
-				msg := fmt.Sprintf("MountVolume.GetHostIDsForPod failed to find host ID in user namespace (UID: %v GID: %v)", containerUID, containerGID)
-				eventErr, detailedErr := volumeToMount.GenerateError(msg, err)
-				return volumetypes.NewOperationContext(eventErr, detailedErr, migrated)
-			}
-		}
-
 		// Execute mount
```

Comment:
So the idea here is to, instead of kubelet giving permissions to mapped uid/gid, we are going to ask CRI to idmap these mount points when creating bind mounts? So no change in volumemanager code is necessary?

Reply:
yes exactly, the idea is to use idmapped mounts (through the CRI) to cancel the effect on volumes of running inside a user namespace